src/definitions/object.ts

```diff
@@ -57,7 +63,7 @@ ${getDataClassMembers({ node, schema, config, completableFuture: true })}
 }`;
   }
 
-  return `${annotations}data class ${name}(
+  return `${annotations}${typeConsolidationAnnotation}data class ${name}(
```

Comment:
Will only add the valid object locations to data classes. The interfaces will have `@GraphQLIgnore` so clients will need to do this themselves